pkg/network/vmispec/interface.go

```diff
@@ -49,16 +49,41 @@ func IsPodNetworkWithMasqueradeBindingInterface(networks []v1.Network, ifaces []
 }
 
 func LookupInterfaceStatusByMac(interfaces []v1.VirtualMachineInstanceNetworkInterface, macAddress string) *v1.VirtualMachineInstanceNetworkInterface {
-	for _, iface := range interfaces {
-		if iface.MAC == macAddress {
-			iface := iface
-			return &iface
```

Comment:
I think the original intention was to avoid pointer to a slice item, it makes sense since the slice is not immutable.
What is the reason for the change?